File: VB/BarManagerHelper.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections
Imports System.Data
Imports System.Diagnostics
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraBars
Imports DevExpress.XtraEditors.Repository

Namespace WindowsApplication41
	Public Class BarManagerHelper
		Private manager As BarManager

		Public Sub New(ByVal manager As BarManager)
			Me.manager = manager
		End Sub

		Public WriteOnly Property SaveEditValueImmediately() As Boolean
			Set(ByVal value As Boolean)
				UnsubscribeFromEvents()
				If value Then
					SubscribeToEvents()
				End If
			End Set
		End Property

		Private Sub SubscribeToEvents()
            For Each item As RepositoryItem In manager.RepositoryItems
                AddHandler item.EditValueChanged, AddressOf OnEditValueChanged
            Next item
		End Sub

		Private Sub UnsubscribeFromEvents()
            For Each item As RepositoryItem In manager.RepositoryItems
                RemoveHandler item.EditValueChanged, AddressOf OnEditValueChanged
            Next item
		End Sub

		Private Sub OnEditValueChanged(ByVal sender As Object, ByVal e As EventArgs)
			manager.ActiveEditItemLink.PostEditor()
		End Sub

	End Class
End Namespace

File: CS/BarManagerHelper.cs
using System;
using System.Collections;
using System.Data;
using System.Diagnostics;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraBars;
using DevExpress.XtraEditors.Repository;

namespace WindowsApplication41
{
    public class BarManagerHelper
    {
        BarManager manager;

        public BarManagerHelper(BarManager manager)
        {
            this.manager = manager;
        }

        public bool SaveEditValueImmediately
        {
            set
            {
                UnsubscribeFromEvents();
                if (value)
                    SubscribeToEvents();
            }
        }

        private void SubscribeToEvents()
        {
            manager.RepositoryItems.CollectionChanged += OnRepositoryItemsCollectionChanged;
            foreach (RepositoryItem item in manager.RepositoryItems)
                item.EditValueChanged += OnEditValueChanged;
        }

        private void UnsubscribeFromEvents()
        {
            manager.RepositoryItems.CollectionChanged -= OnRepositoryItemsCollectionChanged;
            foreach (RepositoryItem item in manager.RepositoryItems)
                item.EditValueChanged -= OnEditValueChanged;
        }

        void OnEditValueChanged(object sender, EventArgs e)
        {
            manager.ActiveEditItemLink.PostEditor();
        }

        void OnRepositoryItemsCollectionChanged(object sender, System.ComponentModel.CollectionChangeEventArgs e)
        {
            RepositoryItem item = e.Element as RepositoryItem;
            if (e.Action == System.ComponentModel.CollectionChangeAction.Add)
                item.EditValueChanged += OnEditValueChanged;
            if (e.Action == System.ComponentModel.CollectionChangeAction.Remove)
                item.EditValueChanged -= OnEditValueChanged;
        }

    }
}
